### PR TITLE
Add `libthermo` and `pythermo` recipes

### PR DESCRIPTION
**Pull request by adriendelsalle**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.

### CODE REVIEW

**Comment by adriendelsalle** [comment]
ping @conda-forge/help-python @conda-forge/help-c-cpp

File: recipes/libthermo/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        - cmake
+        - ninja
+        - xtensor>=0.24
+        - xsimd
+        - tbb-devel
+      host: []  # Empty host dependency section
+
+    test:
+      commands:
+        - test -d ${PREFIX}/include/libthermo  # [unix]
```
Comment (xhochy) [inline recipes/libthermo/meta.yaml:34]: Is this a header-only library?

File: recipes/libthermo/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        - cat $PREFIX/include/libthermo/version.hpp | grep "LIBTHERMO_VERSION_MAJOR {{ version_split[0] }}"  # [unix]
+        - cat $PREFIX/include/libthermo/version.hpp | grep "LIBTHERMO_VERSION_MINOR {{ version_split[1] }}"  # [unix]
+        - cat $PREFIX/include/libthermo/version.hpp | grep "LIBTHERMO_VERSION_PATCH {{ version_split[2] }}"  # [unix]
+
+  - name: pythermo
+    version: {{ version }}
+    script: build_libthermo.sh                       # [unix]
+    script: build_libthermo.bat                      # [win]
+    build:
+      string: py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}
```
Comment (xhochy) [inline recipes/libthermo/meta.yaml:49]: Why do you need to specify this? This looks like the default

File: recipes/libthermo/build_libthermo.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        -DCMAKE_PREFIX_PATH=$PREFIX     \
+        -DCMAKE_BUILD_TYPE="Release"    \
+        -DLIBTHERMO_USE_XTENSOR=ON      \
+        -DXTENSOR_USE_XSIMD=ON          \
+        -DXTENSOR_USE_TBB=ON
+elif [[ $PKG_NAME == "pythermo" ]]; then
+    cmake .. ${CMAKE_ARGS}              \
+        -GNinja                         \
+        -DCMAKE_PREFIX_PATH=$PREFIX     \
+        -DCMAKE_INSTALL_PREFIX=$PREFIX  \
```
Comment (xhochy) [inline recipes/libthermo/build_libthermo.sh:23]: Part of `CMAKE_ARGS`
```suggestion
```

File: recipes/libthermo/build_libthermo.sh [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+if [[ $build_platform == "osx-64" ]]; then
+    set TBB_SWITCH=OFF
+else
+    set TBB_SWITCH=ON
+fi
+if [[ $PKG_NAME == "libthermo" ]]; then
+    cmake .. ${CMAKE_ARGS}              \
+        -GNinja                         \
+        -DCMAKE_INSTALL_PREFIX=$PREFIX  \
+        -DCMAKE_PREFIX_PATH=$PREFIX     \
```
Comment (xhochy) [inline recipes/libthermo/build_libthermo.sh:14]: Part of `CMAKE_ARGS`
```suggestion
```

File: recipes/libthermo/meta.yaml (hunk 1 above)
Comment (adriendelsalle) [reply recipes/libthermo/meta.yaml:34]: yes it is @xhochy

**Review by xhochy — APPROVED** [review]

**Comment by adriendelsalle** [comment]
thx @xhochy !

**Pull request merged**